autode/reactions/reaction_types.py

```diff
@@ -1,16 +1,28 @@
+from typing import Sequence
 from autode.exceptions import ReactionFormationFailed
 from autode.log import logger
 
 
 class ReactionType:
 
-    def __eq__(self, other):
-        return self.name == other.name
-
-    def __init__(self, name):
+    def __init__(self, name: str):
+        """
+        Reaction type with only a name
 
+        -----------------------------------------------------------------------
+        Arguments:
+            name: Name of the reaction type
+        """
         self.name = name
 
+    def __eq__(self, other) -> bool:
+        """Equality of two types is just based on their names"""
+
+        if not isinstance(other, ReactionType):
```

Comment:
`return isinstance(other, ReactionType) and self.name == other.name` ? idk if its better